File: JavaScript/Array/L_minIncrMakeArrEq.js
var minIncrementForUnique = function (nums) {
    nums.sort((a, b) => a - b);

    let numTracker = 0; // Tracks the next unique number that should be set.
    let minIncrement = 0; // Counts the total increments required.

    for (let num of nums) {
        numTracker = Math.max(numTracker, num);
        minIncrement += numTracker - num;
        numTracker++; // Increment the tracker for the next number.
    }

    return minIncrement;
};

File: Python/Arrays/L_minIncrMakeArrEq.py
def minIncrementForUnique(nums):
        nums.sort()
        ans = 0
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                ans += nums[i - 1] - nums[i] + 1
                nums[i] = nums[i - 1] + 1
        return ans
        